# LDED Audiovisual Fusion 

Author: Chen Lequn.
Created on 31 Oct 2023.


### Notebook 4b: ML modelling - ablation study (feature-based audio visual fusion model)
- Full Dataset: All features (from 4a)
- Ablated Dataset 1: Features after removing those with Pearson correlation > 0.9 (within each modality).
- Ablated Dataset 2: Features from Ablated Dataset 1 with MI > 0.05.
- Ablated Dataset 3: Features from Ablated Dataset 1 with MI > 0.1.
- Ablated Dataset 4: Audio Features from All features.
- Ablated Dataset 5: Vision Features from All features.
- Ablated Dataset 6: Audio Features from Ablated Dataset 1.
- Ablated Dataset 7: Vision Features from Ablated Dataset 1.
- Ablated Dataset 8: Audio Features from Ablated Dataset 2.
- Ablated Dataset 9: Audio Features from Ablated Dataset 3.

In [1]:
# from sklearnex import patch_sklearn, config_context
# import dpctl
# patch_sklearn()

In [2]:
import wandb

In [3]:
import os
import glob
import sys
import math
import numpy as np
import pandas as pd
from pandas import DataFrame
import scipy as sp
import itertools
from collections import defaultdict
from pprint import pprint
import pickle
from joblib import dump, load

# Plotting
import matplotlib as mpl
import matplotlib.pyplot as plt
from matplotlib.pyplot import gca
from pylab import *
import seaborn as sns
mpl.rc('axes', labelsize=14)
mpl.rc('xtick', labelsize=12)
mpl.rc('ytick', labelsize=12)
%matplotlib inline

# Scikit-learn general
from sklearn import preprocessing, metrics, svm, datasets, tree, neighbors
from sklearn.metrics import accuracy_score, f1_score, roc_auc_score, precision_score, recall_score, confusion_matrix, roc_curve, auc, classification_report
from sklearn.model_selection import train_test_split, RandomizedSearchCV, cross_validate, KFold, StratifiedKFold, GridSearchCV
from sklearn.preprocessing import StandardScaler, LabelEncoder, OneHotEncoder, scale
from sklearn.ensemble import HistGradientBoostingClassifier

# Scikit-learn models
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.linear_model import LogisticRegression, Lasso, Ridge
from sklearn.ensemble import RandomForestClassifier, AdaBoostClassifier, GradientBoostingClassifier, GradientBoostingRegressor
from sklearn.neural_network import MLPClassifier, MLPRegressor
from sklearn.gaussian_process import GaussianProcessClassifier
from sklearn.gaussian_process.kernels import RBF
from sklearn.naive_bayes import GaussianNB
from sklearn.svm import SVC, SVR
from sklearn.multiclass import OneVsRestClassifier

# Imbalanced learn
from imblearn.ensemble import BalancedRandomForestClassifier, RUSBoostClassifier

# XGBoost and LightGBM
import xgboost as xgb
from xgboost import XGBClassifier
import lightgbm as lgb
from lightgbm import LGBMClassifier

# Utilities
from scipy.stats import uniform, randint
from scipy.interpolate import griddata
from sklearn.utils import shuffle, resample, class_weight
from utils import *


In [4]:
PROJECT_ROOT_DIR = "../"
IMAGE_PATH = os.path.join(PROJECT_ROOT_DIR, "result_images", 'defect classification', 'class_name_v3', 'ML-baseline', 'ablation study')
model_path = os.path.join(PROJECT_ROOT_DIR, "trained_models", 'defect classification', 'class_name_v3', 'ML-baseline', 'ablation study')
scaler_path = os.path.join(PROJECT_ROOT_DIR, "saved_scalers", 'defect classification', 'class_name_v3', 'ML-baseline', 'ablation study')
os.makedirs(IMAGE_PATH, exist_ok=True)
os.makedirs(model_path, exist_ok=True)
os.makedirs(scaler_path, exist_ok=True)

Multimodal_dataset_PATH = "/home/chenlequn/Dataset/LDED_acoustic_visual_monitoring_dataset"
Dataset_path = os.path.join(Multimodal_dataset_PATH, f'25Hz')
                            

## function for automatically save the diagram/graph into the folder 
def save_fig(fig_id, tight_layout=True, fig_extension="png", resolution=300):
    path = os.path.join(IMAGE_PATH, fig_id + "." + fig_extension)
    print("Saving figure", fig_id)
    if tight_layout:
        plt.tight_layout()
    plt.savefig(path, format=fig_extension, dpi=resolution)

import warnings
warnings.filterwarnings(action="ignore", message="^internal gelsd")

plt.rcParams["axes.edgecolor"] = "black"
plt.rcParams["axes.linewidth"] = 2.50

In [5]:
# Initialize wandb
wandb.init(project="AudioVisual_ML_ablation_study_v2")

wandb: Currently logged in as: chen1189 (lequn). Use `wandb login --relogin` to force relogin


## Step 1: Dataset preparation

In [6]:
# Read the data back from the HDF5 file into a new DataFrame
df_multimodal = pd.read_hdf(os.path.join(Dataset_path, 'spatiotemporal_fused_multimodal.h5'), key='df')
df_multimodal = df_multimodal.dropna(subset=['class_name_v3'])
df_multimodal

sample index  Time_Stamps      audio_file_name      image_file_name  \
0                 1         0.00      sample_21_1.wav      sample_21_1.jpg   
1                 2         0.04      sample_21_2.wav      sample_21_2.jpg   
2                 3         0.08      sample_21_3.wav      sample_21_3.jpg   
3                 4         0.12      sample_21_4.wav      sample_21_4.jpg   
4                 5         0.16      sample_21_5.wav      sample_21_5.jpg   
...             ...          ...                  ...                  ...   
48461         10982       439.24  sample_26_10982.wav  sample_26_10982.jpg   
48462         10983       439.28  sample_26_10983.wav  sample_26_10983.jpg   
48463         10984       439.32  sample_26_10984.wav  sample_26_10984.jpg   
48464         10985       439.36  sample_26_10985.wav  sample_26_10985.jpg   
48465         10986       439.40  sample_26_10986.wav  sample_26_10986.jpg   

        class_name class_name_v2  Layer number  Sample number class_name_v3  \
0        Laser-off     Laser-off           1.0             21     Laser-off   
1      Defect-free   Defect-free           1.0             21   Defect-free   
2      Defect-free   Defect-free           1.0             21   Defect-free   
3      Defect-free   Defect-free           1.0             21   Defect-free   
4      Defect-free   Defect-free           1.0             21   Defect-free   
...            ...           ...           ...            ...           ...   
48461    Laser-off     Laser-off           NaN             26     Laser-off   
48462    Laser-off     Laser-off           NaN             26     Laser-off   
48463    Laser-off     Laser-off           NaN             26     Laser-off   
48464    Laser-off     Laser-off           NaN             26     Laser-off   
48465    Laser-off     Laser-off           NaN             26     Laser-off   

       rms_energy  ...    center_y  contour_area  contour_length      X  \
0        0.009018  ...    0.000000           0.0        0.000000  0.000   
1        0.019286  ...  239.000000      305442.0     2234.000000  0.000   
2        0.019593  ...  239.000000      305442.0     2234.000000 -0.010   
3        0.030937  ...  230.264496      291865.5     2279.781744 -0.007   
4        0.038329  ...  229.109962      281970.0     2407.847760 -0.111   
...           ...  ...         ...           ...             ...    ...   
48461    0.011000  ...    0.000000           0.0        0.000000  0.000   
48462    0.008347  ...    0.000000           0.0        0.000000  0.000   
48463    0.011081  ...    0.000000           0.0        0.000000  0.000   
48464    0.011044  ...    0.000000           0.0        0.000000  0.000   
48465    0.013110  ...    0.000000           0.0        0.000000  0.000   

           Y      Z     Vx     Vy     Vz  Speed  
0      0.000  0.000  0.000 -0.000  0.000  0.000  
1     -0.000  0.001 -0.074 -0.074  0.000  0.105  
2      0.015 -0.007 -0.051  0.026 -0.051  0.077  
3      0.016  0.010  0.173 -0.222  2.520  2.536  
4      0.123 -0.057 -1.864  1.662 -2.468  3.511  
...      ...    ...    ...    ...    ...    ...  
48461  0.000  0.000 -0.076  0.025  0.000  0.080  
48462  0.000  0.000 -0.076  0.025  0.000  0.080  
48463  0.000  0.000 -0.076  0.025  0.000  0.080  
48464  0.000  0.000 -0.076  0.025  0.000  0.080  
48465  0.000  0.000 -0.076  0.025  0.000  0.080  

[36557 rows x 135 columns]

In [7]:
# Extract labels
y = df_multimodal['class_name_v3'].to_list()
y_array = np.array(y)
le = LabelEncoder()
le.fit(y)
Y_encoded = le.transform(y)

# Non-feature columns to be dropped
non_feature_columns = ['sample index', 'Time_Stamps', 'audio_file_name', 'image_file_name',
                       'class_name', 'class_name_v2', 'class_name_v3', 'Layer number', 'Sample number', 'X', 'Y', 'Z', 'Vx', 'Vy', 'Vz', 'Speed']

# Extract features for the full dataset
X_all = df_multimodal.drop(columns=non_feature_columns).to_numpy()

# Redundant audio features to be dropped
redundant_audio_features = ['dynamic_complexity','loudness']

# Extract audio features
audio_feature_columns = ['rms_energy', 'amplitude_envelope_mean',
                           'amplitude_envelope_std', 'zero_crossing_rate', 'dynamic_complexity',
                           'loudness', 'loudness_vickers', 'spectral_centroid_mean',
                           'spectral_centroid_std', 'spectral_complexity_mean',
                           'spectral_complexity_std', 'spectral_contrast_0_mean',
                           'spectral_contrast_0_std', 'spectral_contrast_1_mean',
                           'spectral_contrast_1_std', 'spectral_contrast_2_mean',
                           'spectral_contrast_2_std', 'spectral_contrast_3_mean',
                           'spectral_contrast_3_std', 'spectral_contrast_4_mean',
                           'spectral_contrast_4_std', 'spectral_contrast_5_mean',
                           'spectral_contrast_5_std', 'spectral_valley_0_mean',
                           'spectral_valley_0_std', 'spectral_valley_1_mean',
                           'spectral_valley_1_std', 'spectral_valley_2_mean',
                           'spectral_valley_2_std', 'spectral_valley_3_mean',
                           'spectral_valley_3_std', 'spectral_valley_4_mean',
                           'spectral_valley_4_std', 'spectral_valley_5_mean',
                           'spectral_valley_5_std', 'spectral_decrease_mean',
                           'spectral_decrease_std', 'spectral_energy_mean', 'spectral_energy_std',
                           'spectral_energy_band_ratio_mean', 'spectral_energy_band_ratio_std',
                           'spectral_flatness_mean', 'spectral_flatness_std', 'spectral_flux_mean',
                           'spectral_flux_std', 'spectral_rolloff_mean', 'spectral_rolloff_std',
                           'spectral_strong_peak_mean', 'spectral_strong_peak_std',
                           'spectral_variance_mean', 'spectral_variance_std',
                           'spectral_skewness_mean', 'spectral_skewness_std',
                           'spectral_kurtosis_mean', 'spectral_kurtosis_std',
                           'spectral_crest_factor_mean', 'spectral_crest_factor_std',
                           'mfcc_0_mean', 'mfcc_0_std', 'mfcc_1_mean', 'mfcc_1_std', 'mfcc_2_mean',
                           'mfcc_2_std', 'mfcc_3_mean', 'mfcc_3_std', 'mfcc_4_mean', 'mfcc_4_std',
                           'mfcc_5_mean', 'mfcc_5_std', 'mfcc_6_mean', 'mfcc_6_std', 'mfcc_7_mean',
                           'mfcc_7_std', 'mfcc_8_mean', 'mfcc_8_std', 'mfcc_9_mean', 'mfcc_9_std',
                           'mfcc_10_mean', 'mfcc_10_std', 'mfcc_11_mean', 'mfcc_11_std',
                           'mfcc_12_mean', 'mfcc_12_std']

X_audio = df_multimodal[audio_feature_columns].drop(columns=redundant_audio_features).to_numpy()

# Extract vision features
vision_feature_columns = ['max_contour_area', 'rectangle_angle',
                          'rectangle_width', 'rectangle_height', 'ellipse_angle', 'ellipse_width',
                          'ellipse_height', 'max_hull', 'm00', 'm10', 'm01', 'm20', 'm11', 'm02',
                          'm30', 'm21', 'm12', 'm03', 'mu20', 'mu11', 'mu02', 'mu30', 'mu21',
                          'mu12', 'mu03', 'nu20', 'nu11', 'nu02', 'nu30', 'nu21', 'nu12', 'nu03',
                          'center_x', 'center_y', 'contour_area', 'contour_length']

X_vision = df_multimodal[vision_feature_columns].to_numpy()

X_all.shape, X_audio.shape, X_vision.shape

((36557, 119), (36557, 81), (36557, 36))

In [8]:
class_weights = class_weight.compute_class_weight('balanced',
                                                 classes = np.unique(y),
                                                 y = y)

class_weights_encoded = class_weight.compute_class_weight('balanced',
                                                 classes = np.unique(Y_encoded),
                                                 y = Y_encoded)


class_names = (le.classes_).tolist()
print(f"class weights: {class_weights}")
print(f"class weights encoded: {class_weights_encoded}")
print(f"class        : {class_names}")

class weights: [1.20237469 1.83114606 1.472173   0.51468435]
class weights encoded: [1.20237469 1.83114606 1.472173   0.51468435]
class        : ['Defect-free', 'Keyhole pores', 'Laser defocus', 'Laser-off']


In [9]:
# Given features based on previous analysis results
features_dropped_090 = ['loudness_vickers', 'spectral_centroid_mean',
       'spectral_centroid_std', 'spectral_complexity_mean',
       'spectral_complexity_std', 'spectral_contrast_1_mean',
       'spectral_contrast_3_mean', 'spectral_contrast_3_std',
       'spectral_contrast_4_mean', 'spectral_contrast_5_mean',
       'spectral_valley_0_mean', 'spectral_valley_2_mean',
       'spectral_valley_3_mean', 'spectral_valley_4_mean',
       'spectral_valley_5_mean', 'spectral_energy_band_ratio_std',
       'spectral_flux_mean', 'spectral_flux_std', 'spectral_rolloff_mean',
       'spectral_strong_peak_mean', 'spectral_kurtosis_mean',
       'spectral_crest_factor_mean', 'spectral_crest_factor_std',
       'mfcc_1_mean', 'mfcc_2_mean', 'mfcc_3_mean', 'mfcc_4_std',
       'mfcc_6_mean', 'max_contour_area', 'rectangle_angle', 'ellipse_angle',
       'nu02', 'nu03']

features_dropped_090_mi_filtered_01 = ['loudness_vickers', 'spectral_complexity_mean',
       'spectral_complexity_std', 'spectral_contrast_1_mean',
       'spectral_contrast_3_mean', 'spectral_valley_0_mean',
       'spectral_valley_2_mean', 'spectral_valley_3_mean',
       'spectral_valley_4_mean', 'spectral_valley_5_mean',
       'spectral_flux_mean', 'spectral_flux_std', 'spectral_rolloff_mean',
       'spectral_kurtosis_mean', 'mfcc_1_mean', 'mfcc_3_mean',
       'max_contour_area', 'rectangle_angle', 'ellipse_angle', 'nu02', 'nu03']

features_dropped_090_mi_filtered_005 = ['loudness_vickers', 'spectral_centroid_mean',
       'spectral_centroid_std', 'spectral_complexity_mean',
       'spectral_complexity_std', 'spectral_contrast_1_mean',
       'spectral_contrast_3_mean', 'spectral_contrast_3_std',
       'spectral_contrast_4_mean', 'spectral_contrast_5_mean',
       'spectral_valley_0_mean', 'spectral_valley_2_mean',
       'spectral_valley_3_mean', 'spectral_valley_4_mean',
       'spectral_valley_5_mean', 'spectral_energy_band_ratio_std',
       'spectral_flux_mean', 'spectral_flux_std', 'spectral_rolloff_mean',
       'spectral_strong_peak_mean', 'spectral_kurtosis_mean',
       'spectral_crest_factor_mean', 'spectral_crest_factor_std',
       'mfcc_1_mean', 'mfcc_2_mean', 'mfcc_3_mean', 'mfcc_4_std',
       'mfcc_6_mean', 'max_contour_area', 'rectangle_angle', 'ellipse_angle',
       'nu02', 'nu03']

# Create ablated datasets
X_dropped_090 = df_multimodal[features_dropped_090].to_numpy()
X_dropped_090_mi_filtered_01 = df_multimodal[features_dropped_090_mi_filtered_01].to_numpy()
X_dropped_090_mi_filtered_005 = df_multimodal[features_dropped_090_mi_filtered_005].to_numpy()

# Extract Audio and Vision features from the ablated datasets
X_audio_dropped_090 = df_multimodal[[feat for feat in features_dropped_090 if feat in audio_feature_columns]].to_numpy()
X_vision_dropped_090 = df_multimodal[[feat for feat in features_dropped_090 if feat in vision_feature_columns]].to_numpy()

# Extract Audio features from the ablated datasets 2 and 3
X_audio_dropped_090_mi_filtered_01 = df_multimodal[[feat for feat in features_dropped_090_mi_filtered_01 if feat in audio_feature_columns]].to_numpy()
X_audio_dropped_090_mi_filtered_005 = df_multimodal[[feat for feat in features_dropped_090_mi_filtered_005 if feat in audio_feature_columns]].to_numpy()

# Get the shapes of these additional ablated datasets
X_dropped_090.shape, X_dropped_090_mi_filtered_01.shape, X_dropped_090_mi_filtered_005.shape, X_audio_dropped_090.shape, X_vision_dropped_090.shape, X_audio_dropped_090_mi_filtered_01.shape, X_audio_dropped_090_mi_filtered_005.shape

((36557, 33),
 (36557, 21),
 (36557, 33),
 (36557, 28),
 (36557, 5),
 (36557, 16),
 (36557, 28))

- ALL Dataset (X_all):  `119 features`
- Ablated Dataset 1 (X_dropped_090):  `33 features`
- Ablated Dataset 2 (X_dropped_090_mi_filtered_01): `21 features`
- Ablated Dataset 3 (X_dropped_090_mi_filtered_005):  `33 features`
- Ablated Dataset 4 (X_audio_all): `81 audio features`
- Ablated Dataset 5 (X_vision_all): `36 vision features`
- Ablated Dataset 6 (X_audio_dropped_090): `28 audio features`
- Ablated Dataset 7 (X_vision_dropped_090): `5 vision features`
- Ablated Dataset 8 (X_audio_dropped_090_mi_filtered_01): `16 audio features`
- Ablated Dataset 9 (X_audio_dropped_090_mi_filtered_005): `28 audio features`

In [10]:
%store -r train_indices
%store -r val_indices
%store -r test_indices

In [11]:
len(test_indices), len(train_indices)

(462, 3692)

## Step 2: Create a Pipeline and Iterate Over Datasets and Models

In [12]:
# Define hyperparameters for each model
hyperparameters = {
    'KNN': {'n_neighbors': 12, 'weights': 'distance'},
    'DT': {'ccp_alpha': 3.89e-05, 'max_depth': 21, 'min_samples_leaf': 1, 'min_samples_split': 13},
    'LR': {'C': 5.15, 'max_iter': 2528, 'penalty': 'l2', 'solver': 'lbfgs'},
    'RF': {'bootstrap': False, 'max_depth': 64, 'max_features': 'sqrt', 'min_samples_leaf': 3, 'min_samples_split': 6, 'n_estimators': 307},
    'BalancedRF': {'bootstrap': False, 'max_depth': 23, 'max_features': 'sqrt', 'min_samples_leaf': 8, 'min_samples_split': 17, 'n_estimators': 237},
    'RUSBoost': {'learning_rate': 0.1, 'n_estimators': 150},
    'Ada': {'algorithm': 'SAMME', 'learning_rate': 0.57, 'n_estimators': 307},
    'GP': {'kernel': 1**2 * RBF(length_scale=1)},
    'NN': {'activation': 'relu', 'alpha': 0.077, 'early_stopping': True, 'hidden_layer_sizes': (229, 255, 168), 'learning_rate': 'adaptive', 'max_iter': 11805, 'solver': 'adam', 'validation_fraction': 0.1},
    'XGBoost': {'colsample_bytree': 0.62, 'gamma': 0.48, 'learning_rate': 0.35, 'max_depth': 44, 'min_child_weight': 8, 'n_estimators': 480, 'subsample': 0.59,  'gpu_id': 0},
    'LightGBM': {'colsample_bytree': 0.80, 'learning_rate': 0.15, 'max_depth': 35, 'min_child_weight': 1, 'n_estimators': 276, 'num_leaves': 120, 'reg_alpha': 1.94, 'reg_lambda': 1.70, 'subsample': 0.86, 'verbose': -1},
    'SVM': {'C': 524.86, 'degree': 9, 'gamma': 0.029, 'kernel': 'rbf', 'probability': True},
    'GB': {'learning_rate': 0.23, 'max_depth': 49, 'min_samples_leaf': 10, 'min_samples_split': 17, 'n_estimators': 271, 'n_iter_no_change': 10, 'subsample': 0.73, 'tol': 0.0001},
    'HistGB': {'learning_rate': 0.23, 'max_iter': 100, 'max_depth': 49, 'min_samples_leaf': 10, 'n_iter_no_change': 10, 'tol': 0.0001},
}

# List of datasets
datasets = {
    'All Features': X_all,
    'Ablated 1': X_dropped_090,
    'Ablated 2': X_dropped_090_mi_filtered_01,
    'Ablated 3': X_dropped_090_mi_filtered_005,
    'Audio All': X_audio,
    'Vision All': X_vision,
    'Audio Ablated 1': X_audio_dropped_090,
    'Vision Ablated 1': X_vision_dropped_090,
    'Audio Ablated 2': X_audio_dropped_090_mi_filtered_01,
    'Audio Ablated 3': X_audio_dropped_090_mi_filtered_005
}


# List of models
models = {
    'KNN': KNeighborsClassifier,
    'DT': DecisionTreeClassifier,
    # 'LR': LogisticRegression,
    'RF': RandomForestClassifier,
    'BalancedRF': BalancedRandomForestClassifier,
    'RUSBoost': RUSBoostClassifier,
    'Ada': AdaBoostClassifier,
    # 'GP': GaussianProcessClassifier,
    'NN': MLPClassifier,
    'XGBoost': XGBClassifier,
    'LightGBM': LGBMClassifier,
    'SVM': SVC,
    # 'GB': GradientBoostingClassifier,
    'HistGB': HistGradientBoostingClassifier
}

## Step 3: Model Training

In [13]:
# Initialize KFold with the number of splits you desire
kf = KFold(n_splits=5)
# Create a dictionary to store final results
final_results = defaultdict(list)
# Create a dictionary to store temporary cross-validation results for each dataset and model
cv_results = defaultdict(lambda: defaultdict(list))


# Iterate over datasets and models
for dataset_name, X in datasets.items():
    print(f"Processing dataset: {dataset_name}")
    # Split data
    X_train = X[train_indices]
    X_val = X[val_indices]
    X_test = X[test_indices]
  
    y_train = Y_encoded[train_indices]
    y_val = Y_encoded[val_indices]
    y_test = Y_encoded[test_indices]

    # Scaling
    scaler = StandardScaler()
    X_train = scaler.fit_transform(X_train)
    X_val = scaler.transform(X_val)
    X_test = scaler.transform(X_test)
    
    with open(os.path.join(scaler_path, f'StandardScaler_{dataset_name}.pkl'), 'wb') as file:
        pickle.dump(scaler, file)

    for model_name, Model in models.items():
        print(f"\tTraining model: {model_name}")
        with wandb.init(project="AudioVisual_ML_ablation_study_v2", name=f"{model_name}_{dataset_name}", reinit=True) as run:
            # Log model and hyperparameters
            wandb.config.update({"Model": model_name, "Hyperparameters": hyperparameters[model_name]})
    
            # Initialize model with the best hyperparameters
            model = Model(**hyperparameters[model_name])
    
            # If the model is SVM, use only 20% of the training data
            if model_name == 'SVM':
                sample_indices = np.random.choice(len(X_train), int(0.2 * len(X_train)), replace=False)
                X_train_sampled = X_train[sample_indices]
                y_train_sampled = y_train[sample_indices]
                model.fit(X_train_sampled, y_train_sampled)
            else:
                # Train the model
                model.fit(X_train, y_train)

            # Save the model
            dump(model, os.path.join(model_path, f'{model_name}_{dataset_name}.joblib'))


            #####------   cross-validation on the testing dataset--------------
            fold_metrics = defaultdict(list)
    
            for train_idx, test_idx in kf.split(X_test):
                X_test_train, X_test_val = X_test[train_idx], X_test[test_idx]
                y_test_train, y_test_val = y_test[train_idx], y_test[test_idx]
                
                # Make predictions
                y_test_pred = model.predict(X_test_val)
                
                # Evaluate the model
                acc = accuracy_score(y_test_val, y_test_pred)
                f1 = f1_score(y_test_val, y_test_pred, average='weighted')
                precision = precision_score(y_test_val, y_test_pred, average='weighted')
                recall = recall_score(y_test_val, y_test_pred, average='weighted')
                
                # Optional: ROC-AUC can only be calculated for binary classification or multilabel indicator format
                try:
                    roc_auc = roc_auc_score(y_test_val, model.predict_proba(X_test_val), multi_class='ovr', average='weighted')
                    
                except:
                    roc_auc = 'N/A'
                
                # Store the metrics for this fold
                fold_metrics['Accuracy'].append(acc)
                fold_metrics['F1 Score'].append(f1)
                fold_metrics['Precision'].append(precision)
                fold_metrics['Recall'].append(recall)
                fold_metrics['ROC-AUC'].append(roc_auc if roc_auc != 'N/A' else np.nan)
            
            # Calculate mean and std for each metric and print them
            for metric, values in fold_metrics.items():
                mean_val = np.nanmean(values)  # Handle 'N/A' values
                std_val = np.nanstd(values)    # Handle 'N/A' values
                
                print(f"\t\t{metric}: Mean = {mean_val}, Std = {std_val}")
                # Log the mean and std values to wandb
                wandb.log({f"{metric}_mean": mean_val, f"{metric}_std": std_val})
                
                cv_results[dataset_name][f"{model_name}_{metric}_mean"] = mean_val
                cv_results[dataset_name][f"{model_name}_{metric}_std"] = std_val
             
            #####------   cross-validation on the testing dataset--------------
            
# Convert the temporary results to final format
for dataset_name, metrics in cv_results.items():
    final_results['Dataset'].append(dataset_name)
    for metric_name, value in metrics.items():
        final_results[metric_name].append(value)

# Convert results to a DataFrame for easier viewing
final_results_df = pd.DataFrame(final_results)

Processing dataset: All Features
	Training model: KNN


		Accuracy: Mean = 0.8701963534361852, Std = 0.04364919371953449
		F1 Score: Mean = 0.872004096850047, Std = 0.04237379402456881
		Precision: Mean = 0.8787829247671806, Std = 0.03908269162088463
		Recall: Mean = 0.8701963534361852, Std = 0.04364919371953449
		ROC-AUC: Mean = 0.9692253336554998, Std = 0.013861821271084847


	Training model: DT


		Accuracy: Mean = 0.8442730247779335, Std = 0.04490565142150513
		F1 Score: Mean = 0.8458100105448153, Std = 0.044475053765800146
		Precision: Mean = 0.8504462469735087, Std = 0.0442976986779252
		Recall: Mean = 0.8442730247779335, Std = 0.04490565142150513
		ROC-AUC: Mean = 0.8995609118256727, Std = 0.0314111711366799


	Training model: RF


		Accuracy: Mean = 0.9069892473118278, Std = 0.014229539075207975
		F1 Score: Mean = 0.9070118062491442, Std = 0.013415782027742318
		Precision: Mean = 0.9115972647824908, Std = 0.011923419858057498
		Recall: Mean = 0.9069892473118278, Std = 0.014229539075207975
		ROC-AUC: Mean = 0.9851137803065562, Std = 0.004255251048746643


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.8550257129499765, Std = 0.034467244781384465
		F1 Score: Mean = 0.8560736138440322, Std = 0.03284455423026473
		Precision: Mean = 0.8613628916149194, Std = 0.029465718527128017
		Recall: Mean = 0.8550257129499765, Std = 0.034467244781384465
		ROC-AUC: Mean = 0.9728304161598377, Std = 0.008689111501593365


	Training model: RUSBoost


		Accuracy: Mean = 0.8441795231416551, Std = 0.03320932445984241
		F1 Score: Mean = 0.8451444646782551, Std = 0.033239432782763
		Precision: Mean = 0.8547187360389156, Std = 0.029186876296783237
		Recall: Mean = 0.8441795231416551, Std = 0.03320932445984241
		ROC-AUC: Mean = 0.9389214490064515, Std = 0.013661218917655118


	Training model: Ada


		Accuracy: Mean = 0.863627863487611, Std = 0.04136413211370029
		F1 Score: Mean = 0.8662772996490988, Std = 0.03917377774677887
		Precision: Mean = 0.8791436669522676, Std = 0.03763067730564038
		Recall: Mean = 0.863627863487611, Std = 0.04136413211370029
		ROC-AUC: Mean = 0.9540318947172143, Std = 0.006306090865851742


	Training model: NN


		Accuracy: Mean = 0.9199625993454885, Std = 0.02203092312041603
		F1 Score: Mean = 0.9198541353213268, Std = 0.02108884316200871
		Precision: Mean = 0.9231193004882963, Std = 0.020026249055532578
		Recall: Mean = 0.9199625993454885, Std = 0.02203092312041603
		ROC-AUC: Mean = 0.9845578006420261, Std = 0.00622152121260869


	Training model: XGBoost


		Accuracy: Mean = 0.9198690977092099, Std = 0.031294827951081544
		F1 Score: Mean = 0.9196315244992157, Std = 0.030369019043249103
		Precision: Mean = 0.922995659748211, Std = 0.030266932246708227
		Recall: Mean = 0.9198690977092099, Std = 0.031294827951081544
		ROC-AUC: Mean = 0.989299498331982, Std = 0.0024162296317778165


	Training model: LightGBM


		Accuracy: Mean = 0.9177653108929406, Std = 0.031087764956929994
		F1 Score: Mean = 0.9178823358034356, Std = 0.03040725621005177
		Precision: Mean = 0.9209861621211258, Std = 0.029451333304741305
		Recall: Mean = 0.9177653108929406, Std = 0.031087764956929994
		ROC-AUC: Mean = 0.9888707206323945, Std = 0.0034436208101405645


	Training model: SVM


		Accuracy: Mean = 0.8767414679756896, Std = 0.03480861354268683
		F1 Score: Mean = 0.8776921406344294, Std = 0.03307198992993354
		Precision: Mean = 0.8832507447754947, Std = 0.03012307329065782
		Recall: Mean = 0.8767414679756896, Std = 0.03480861354268683
		ROC-AUC: Mean = 0.9703731724962585, Std = 0.012454497802001092


	Training model: HistGB


		Accuracy: Mean = 0.9329593267882188, Std = 0.034414175590850686
		F1 Score: Mean = 0.933320710307977, Std = 0.03332342930032505
		Precision: Mean = 0.9365152444999163, Std = 0.03164710686030576
		Recall: Mean = 0.9329593267882188, Std = 0.034414175590850686
		ROC-AUC: Mean = 0.9883410451190491, Std = 0.004232936879810858


Processing dataset: Ablated 1
	Training model: KNN


		Accuracy: Mean = 0.874520804114072, Std = 0.024975352966933692
		F1 Score: Mean = 0.8743344123598868, Std = 0.024516346128526382
		Precision: Mean = 0.8785796289422965, Std = 0.024317623448871704
		Recall: Mean = 0.874520804114072, Std = 0.024975352966933692
		ROC-AUC: Mean = 0.9612187918303903, Std = 0.012531442361516834


	Training model: DT


		Accuracy: Mean = 0.8420289855072463, Std = 0.02407900876919361
		F1 Score: Mean = 0.8432755238134686, Std = 0.024671150472418836
		Precision: Mean = 0.8509042900951582, Std = 0.024248413809903127
		Recall: Mean = 0.8420289855072463, Std = 0.02407900876919361
		ROC-AUC: Mean = 0.9152906325077478, Std = 0.019587370597239397


	Training model: RF


		Accuracy: Mean = 0.8917484805984104, Std = 0.024884481633796893
		F1 Score: Mean = 0.8915117036680789, Std = 0.02355051008270435
		Precision: Mean = 0.8975273726303319, Std = 0.022755733961162948
		Recall: Mean = 0.8917484805984104, Std = 0.024884481633796893
		ROC-AUC: Mean = 0.9818999335767504, Std = 0.0058849584174839945


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.8680691912108461, Std = 0.04161738031727703
		F1 Score: Mean = 0.8686504728368959, Std = 0.04059482712352724
		Precision: Mean = 0.8743534759158941, Std = 0.03721064074531755
		Recall: Mean = 0.8680691912108461, Std = 0.04161738031727703
		ROC-AUC: Mean = 0.9747095746073583, Std = 0.009835624024106024


	Training model: RUSBoost


		Accuracy: Mean = 0.8269752220663861, Std = 0.044962622131493685
		F1 Score: Mean = 0.8305368400850448, Std = 0.0435772943183485
		Precision: Mean = 0.8469311622225796, Std = 0.03698082659890567
		Recall: Mean = 0.8269752220663861, Std = 0.044962622131493685
		ROC-AUC: Mean = 0.9396914994037677, Std = 0.017874308869212374


	Training model: Ada


		Accuracy: Mean = 0.8420991117344553, Std = 0.043291484787212696
		F1 Score: Mean = 0.843857607960628, Std = 0.04191400436367386
		Precision: Mean = 0.856031273590528, Std = 0.03305889032571727
		Recall: Mean = 0.8420991117344553, Std = 0.043291484787212696
		ROC-AUC: Mean = 0.9347136746267001, Std = 0.013096519829728068


	Training model: NN


		Accuracy: Mean = 0.9242870500233753, Std = 0.037494068708444586
		F1 Score: Mean = 0.9248465153724439, Std = 0.03678079802913704
		Precision: Mean = 0.9280601257607394, Std = 0.03532656823519986
		Recall: Mean = 0.9242870500233753, Std = 0.037494068708444586
		ROC-AUC: Mean = 0.9868705662579009, Std = 0.009176839346313977


	Training model: XGBoost


		Accuracy: Mean = 0.9199625993454884, Std = 0.017389387710017728
		F1 Score: Mean = 0.9195041405933271, Std = 0.016706049718945605
		Precision: Mean = 0.9225405542221562, Std = 0.016991193692168335
		Recall: Mean = 0.9199625993454884, Std = 0.017389387710017728
		ROC-AUC: Mean = 0.9863168863825769, Std = 0.005528179160136851


	Training model: LightGBM


		Accuracy: Mean = 0.9156147732585319, Std = 0.023080064999304633
		F1 Score: Mean = 0.9153085870326537, Std = 0.022467538120576853
		Precision: Mean = 0.9174657500574032, Std = 0.023356807991051505
		Recall: Mean = 0.9156147732585319, Std = 0.023080064999304633
		ROC-AUC: Mean = 0.9876381677674898, Std = 0.004432124432249788


	Training model: SVM


		Accuracy: Mean = 0.8918186068256194, Std = 0.0348991491209611
		F1 Score: Mean = 0.8910779576144776, Std = 0.034568669571857565
		Precision: Mean = 0.8948793236654474, Std = 0.03615933906691347
		Recall: Mean = 0.8918186068256194, Std = 0.0348991491209611
		ROC-AUC: Mean = 0.9766407571568461, Std = 0.008286271660978499


	Training model: HistGB


		Accuracy: Mean = 0.9177653108929406, Std = 0.02122837230912141
		F1 Score: Mean = 0.917351275857297, Std = 0.020391727567557676
		Precision: Mean = 0.9202993667792793, Std = 0.021168112829746965
		Recall: Mean = 0.9177653108929406, Std = 0.02122837230912141
		ROC-AUC: Mean = 0.9860826830188936, Std = 0.004170727316100153


Processing dataset: Ablated 2
	Training model: KNN


		Accuracy: Mean = 0.8550490883590462, Std = 0.02669410308545048
		F1 Score: Mean = 0.8562038914199036, Std = 0.02583503161326923
		Precision: Mean = 0.8615829017219887, Std = 0.027101581815487986
		Recall: Mean = 0.8550490883590462, Std = 0.02669410308545048
		ROC-AUC: Mean = 0.9564344930576375, Std = 0.013000163569874709


	Training model: DT


		Accuracy: Mean = 0.8376811594202899, Std = 0.05089086930747852
		F1 Score: Mean = 0.8384916796163768, Std = 0.052318580503465106
		Precision: Mean = 0.8435031359354207, Std = 0.054143333487856625
		Recall: Mean = 0.8376811594202899, Std = 0.05089086930747852
		ROC-AUC: Mean = 0.9076481991403462, Std = 0.020886278745954154


	Training model: RF


		Accuracy: Mean = 0.9026180458158016, Std = 0.034273435782184405
		F1 Score: Mean = 0.9025829623847065, Std = 0.03450330856895442
		Precision: Mean = 0.9080291670721792, Std = 0.03457775226373503
		Recall: Mean = 0.9026180458158016, Std = 0.034273435782184405
		ROC-AUC: Mean = 0.9789294494386416, Std = 0.007053364702731942


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.8572463768115941, Std = 0.04203471846185124
		F1 Score: Mean = 0.858083919816244, Std = 0.03906056794789252
		Precision: Mean = 0.8636614376480332, Std = 0.034947148790041696
		Recall: Mean = 0.8572463768115941, Std = 0.04203471846185124
		ROC-AUC: Mean = 0.9715377100288727, Std = 0.009644664512002047


	Training model: RUSBoost


		Accuracy: Mean = 0.7966339410939691, Std = 0.03521775156866642
		F1 Score: Mean = 0.7991479964529623, Std = 0.03571177755698334
		Precision: Mean = 0.8131774918123412, Std = 0.03356260562049475
		Recall: Mean = 0.7966339410939691, Std = 0.03521775156866642
		ROC-AUC: Mean = 0.9227915129517296, Std = 0.02221980655393765


	Training model: Ada


		Accuracy: Mean = 0.831229546517064, Std = 0.02392116522506494
		F1 Score: Mean = 0.834289763869727, Std = 0.023566492363657647
		Precision: Mean = 0.8439880605931714, Std = 0.02307809332265687
		Recall: Mean = 0.831229546517064, Std = 0.02392116522506494
		ROC-AUC: Mean = 0.9287650346473055, Std = 0.011805128918134143


	Training model: NN


		Accuracy: Mean = 0.904721832632071, Std = 0.04040325689453686
		F1 Score: Mean = 0.9039904601360631, Std = 0.03996008324465928
		Precision: Mean = 0.9057421848798137, Std = 0.04074696029142833
		Recall: Mean = 0.904721832632071, Std = 0.04040325689453686
		ROC-AUC: Mean = 0.9782915608920433, Std = 0.010047630465744894


	Training model: XGBoost


		Accuracy: Mean = 0.898293595137915, Std = 0.03535969318155434
		F1 Score: Mean = 0.8978739073743593, Std = 0.035131017868795356
		Precision: Mean = 0.9023075926803885, Std = 0.033886856251474515
		Recall: Mean = 0.898293595137915, Std = 0.03535969318155434
		ROC-AUC: Mean = 0.9830502056625192, Std = 0.005880857691746197


	Training model: LightGBM


		Accuracy: Mean = 0.9026180458158016, Std = 0.028259835150599655
		F1 Score: Mean = 0.9024484752388959, Std = 0.02773331140264083
		Precision: Mean = 0.9066006303280894, Std = 0.02788122905064029
		Recall: Mean = 0.9026180458158016, Std = 0.028259835150599655
		ROC-AUC: Mean = 0.9817187707521517, Std = 0.00711477486712043


	Training model: SVM


		Accuracy: Mean = 0.852805049088359, Std = 0.04043334989695092
		F1 Score: Mean = 0.8509403282325264, Std = 0.04126414637576182
		Precision: Mean = 0.856082766448322, Std = 0.04221224046714841
		Recall: Mean = 0.852805049088359, Std = 0.04043334989695092
		ROC-AUC: Mean = 0.9633700114143877, Std = 0.019887366195509168


	Training model: HistGB


		Accuracy: Mean = 0.9048620850864889, Std = 0.03941117538584906
		F1 Score: Mean = 0.9045413643555815, Std = 0.039816904527080046
		Precision: Mean = 0.908256046508472, Std = 0.04105573357565269
		Recall: Mean = 0.9048620850864889, Std = 0.03941117538584906
		ROC-AUC: Mean = 0.9848743995224123, Std = 0.005821936580561933


Processing dataset: Ablated 3
	Training model: KNN


		Accuracy: Mean = 0.874520804114072, Std = 0.024975352966933692
		F1 Score: Mean = 0.8743344123598868, Std = 0.024516346128526382
		Precision: Mean = 0.8785796289422965, Std = 0.024317623448871704
		Recall: Mean = 0.874520804114072, Std = 0.024975352966933692
		ROC-AUC: Mean = 0.9612187918303903, Std = 0.012531442361516834


	Training model: DT


		Accuracy: Mean = 0.8290556334735857, Std = 0.03430086212724362
		F1 Score: Mean = 0.8300685351676759, Std = 0.03565830040334056
		Precision: Mean = 0.8394510522328875, Std = 0.03150655347204069
		Recall: Mean = 0.8290556334735857, Std = 0.03430086212724362
		ROC-AUC: Mean = 0.904512718493909, Std = 0.023511951311138875


	Training model: RF


		Accuracy: Mean = 0.8917718560074801, Std = 0.02750402721686938
		F1 Score: Mean = 0.8918392249319662, Std = 0.026787730943851357
		Precision: Mean = 0.8973694593278815, Std = 0.025720996136875406
		Recall: Mean = 0.8917718560074801, Std = 0.02750402721686938
		ROC-AUC: Mean = 0.9826825840764177, Std = 0.006253046368719198


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.865895278167368, Std = 0.03998230989749215
		F1 Score: Mean = 0.8669046793872104, Std = 0.03804111253066339
		Precision: Mean = 0.8714058646701662, Std = 0.0347417909479754
		Recall: Mean = 0.865895278167368, Std = 0.03998230989749215
		ROC-AUC: Mean = 0.9753059667813548, Std = 0.009039668724751648


	Training model: RUSBoost


		Accuracy: Mean = 0.8357176250584384, Std = 0.05427901861142225
		F1 Score: Mean = 0.8382105674317423, Std = 0.053285516234222695
		Precision: Mean = 0.8536289183296901, Std = 0.047562760874982915
		Recall: Mean = 0.8357176250584384, Std = 0.05427901861142225
		ROC-AUC: Mean = 0.9375273980209406, Std = 0.019777877971073433


	Training model: Ada


		Accuracy: Mean = 0.8420991117344553, Std = 0.043291484787212696
		F1 Score: Mean = 0.843857607960628, Std = 0.04191400436367386
		Precision: Mean = 0.856031273590528, Std = 0.03305889032571727
		Recall: Mean = 0.8420991117344553, Std = 0.043291484787212696
		ROC-AUC: Mean = 0.9347136746267001, Std = 0.013096519829728068


	Training model: NN


		Accuracy: Mean = 0.9242402992052362, Std = 0.027427374639566883
		F1 Score: Mean = 0.9235948551067331, Std = 0.027098275971719727
		Precision: Mean = 0.9269054567564489, Std = 0.02743638485700853
		Recall: Mean = 0.9242402992052362, Std = 0.027427374639566883
		ROC-AUC: Mean = 0.9888935970150623, Std = 0.006451799699544579


	Training model: XGBoost


		Accuracy: Mean = 0.9199625993454884, Std = 0.017389387710017728
		F1 Score: Mean = 0.9195041405933271, Std = 0.016706049718945605
		Precision: Mean = 0.9225405542221562, Std = 0.016991193692168335
		Recall: Mean = 0.9199625993454884, Std = 0.017389387710017728
		ROC-AUC: Mean = 0.9863168863825769, Std = 0.005528179160136851


	Training model: LightGBM


		Accuracy: Mean = 0.9156147732585319, Std = 0.023080064999304633
		F1 Score: Mean = 0.9153085870326537, Std = 0.022467538120576853
		Precision: Mean = 0.9174657500574032, Std = 0.023356807991051505
		Recall: Mean = 0.9156147732585319, Std = 0.023080064999304633
		ROC-AUC: Mean = 0.9876381677674898, Std = 0.004432124432249788


	Training model: SVM


		Accuracy: Mean = 0.8766713417484805, Std = 0.026005118951930466
		F1 Score: Mean = 0.8761695483478906, Std = 0.026155028305665125
		Precision: Mean = 0.8785593968926134, Std = 0.026175590129389806
		Recall: Mean = 0.8766713417484805, Std = 0.026005118951930466
		ROC-AUC: Mean = 0.9744868386016942, Std = 0.006584296462421093


	Training model: HistGB


		Accuracy: Mean = 0.9177653108929406, Std = 0.02122837230912141
		F1 Score: Mean = 0.917351275857297, Std = 0.020391727567557676
		Precision: Mean = 0.9202993667792793, Std = 0.021168112829746965
		Recall: Mean = 0.9177653108929406, Std = 0.02122837230912141
		ROC-AUC: Mean = 0.9860826830188936, Std = 0.004170727316100153


Processing dataset: Audio All
	Training model: KNN


		Accuracy: Mean = 0.8183029453015427, Std = 0.05818907020782988
		F1 Score: Mean = 0.8065205682589915, Std = 0.06074486561264875
		Precision: Mean = 0.8121759660687948, Std = 0.05669919447915263
		Recall: Mean = 0.8183029453015427, Std = 0.05818907020782988
		ROC-AUC: Mean = 0.9461432996331446, Std = 0.02142336263292495


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: DT


		Accuracy: Mean = 0.7273024777933614, Std = 0.022599366765284023
		F1 Score: Mean = 0.7330398287954871, Std = 0.02735383724642675
		Precision: Mean = 0.751086591336479, Std = 0.04030242286996655
		Recall: Mean = 0.7273024777933614, Std = 0.022599366765284023
		ROC-AUC: Mean = 0.8317582656436604, Std = 0.013920605476771358


	Training model: RF


		Accuracy: Mean = 0.8139083683964469, Std = 0.055199156424326996
		F1 Score: Mean = 0.8025757013575507, Std = 0.050234275334932396
		Precision: Mean = 0.808456230166066, Std = 0.047056904147049716
		Recall: Mean = 0.8139083683964469, Std = 0.055199156424326996
		ROC-AUC: Mean = 0.9629209233176486, Std = 0.014043828398407223


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.7922861150070126, Std = 0.057217892578123764
		F1 Score: Mean = 0.7952528091810728, Std = 0.060357355276563794
		Precision: Mean = 0.8088563570599943, Std = 0.06714174521747356
		Recall: Mean = 0.7922861150070126, Std = 0.057217892578123764
		ROC-AUC: Mean = 0.950163799223709, Std = 0.015837979439902027


	Training model: RUSBoost


		Accuracy: Mean = 0.7100514258999533, Std = 0.046397413666569846
		F1 Score: Mean = 0.7267928686363839, Std = 0.05614617568257104
		Precision: Mean = 0.7574527022900666, Std = 0.07369964471685264
		Recall: Mean = 0.7100514258999533, Std = 0.046397413666569846
		ROC-AUC: Mean = 0.9080497085523034, Std = 0.02489720434376345


	Training model: Ada


		Accuracy: Mean = 0.7599812996727443, Std = 0.06870559626378073
		F1 Score: Mean = 0.749793744213178, Std = 0.07009433591278796
		Precision: Mean = 0.7575098882728054, Std = 0.06650914586662056
		Recall: Mean = 0.7599812996727443, Std = 0.06870559626378073
		ROC-AUC: Mean = 0.9200703134906139, Std = 0.01834687898080332


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: NN


		Accuracy: Mean = 0.8637213651238896, Std = 0.0369491210030211
		F1 Score: Mean = 0.8614976456306982, Std = 0.033696452330773057
		Precision: Mean = 0.8713572450673741, Std = 0.03808754305324044
		Recall: Mean = 0.8637213651238896, Std = 0.0369491210030211
		ROC-AUC: Mean = 0.9730066640657187, Std = 0.011892987809720416


	Training model: XGBoost


		Accuracy: Mean = 0.8506545114539504, Std = 0.04024254376198817
		F1 Score: Mean = 0.8482511299150689, Std = 0.039753655909299
		Precision: Mean = 0.8518918940908617, Std = 0.042072934667644224
		Recall: Mean = 0.8506545114539504, Std = 0.04024254376198817
		ROC-AUC: Mean = 0.9719225202248086, Std = 0.011583184572758169


	Training model: LightGBM


		Accuracy: Mean = 0.8615708274894811, Std = 0.03612908246918264
		F1 Score: Mean = 0.8558697980902773, Std = 0.03477494372576302
		Precision: Mean = 0.8631987685197317, Std = 0.03644930755971354
		Recall: Mean = 0.8615708274894811, Std = 0.03612908246918264
		ROC-AUC: Mean = 0.9722511286545655, Std = 0.01430241108765057


	Training model: SVM


		Accuracy: Mean = 0.8333800841514727, Std = 0.023992395066316296
		F1 Score: Mean = 0.8234156796377323, Std = 0.02361171682661116
		Precision: Mean = 0.8393921716948901, Std = 0.029063989323492728
		Recall: Mean = 0.8333800841514727, Std = 0.023992395066316296
		ROC-AUC: Mean = 0.9618721800011014, Std = 0.016413233876867012


	Training model: HistGB


		Accuracy: Mean = 0.8658485273492286, Std = 0.03093162873892134
		F1 Score: Mean = 0.8609370108149449, Std = 0.026350538455435318
		Precision: Mean = 0.8685164298149578, Std = 0.03423404759214379
		Recall: Mean = 0.8658485273492286, Std = 0.03093162873892134
		ROC-AUC: Mean = 0.9771695217663993, Std = 0.011806827842014202


Processing dataset: Vision All
	Training model: KNN


		Accuracy: Mean = 0.8030388031790556, Std = 0.030099631201257887
		F1 Score: Mean = 0.8046571457765108, Std = 0.028253641975963834
		Precision: Mean = 0.8112699024855978, Std = 0.025675409593361464
		Recall: Mean = 0.8030388031790556, Std = 0.030099631201257887
		ROC-AUC: Mean = 0.9353725179355274, Std = 0.00985599311017718


	Training model: DT


		Accuracy: Mean = 0.7857176250584386, Std = 0.02098736861459898
		F1 Score: Mean = 0.7857064218343737, Std = 0.019892818158747713
		Precision: Mean = 0.7933449788060125, Std = 0.022733979583738843
		Recall: Mean = 0.7857176250584386, Std = 0.02098736861459898
		ROC-AUC: Mean = 0.8977652513811922, Std = 0.019253819980784702


	Training model: RF


		Accuracy: Mean = 0.8571061243571764, Std = 0.021149979187216554
		F1 Score: Mean = 0.8586559078660002, Std = 0.019592215932682473
		Precision: Mean = 0.8642304722837462, Std = 0.01615288070017582
		Recall: Mean = 0.8571061243571764, Std = 0.021149979187216554
		ROC-AUC: Mean = 0.9671886233116995, Std = 0.007338149346306444


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.828962131837307, Std = 0.031750621178125805
		F1 Score: Mean = 0.829646726449383, Std = 0.029710738742022718
		Precision: Mean = 0.8353143762729051, Std = 0.028136702215821475
		Recall: Mean = 0.828962131837307, Std = 0.031750621178125805
		ROC-AUC: Mean = 0.9572988221887124, Std = 0.004839796020220484


	Training model: RUSBoost


		Accuracy: Mean = 0.6927769985974755, Std = 0.04371411494656933
		F1 Score: Mean = 0.6915950985776419, Std = 0.04290033550626937
		Precision: Mean = 0.7289775861560766, Std = 0.037653870296755036
		Recall: Mean = 0.6927769985974755, Std = 0.04371411494656933
		ROC-AUC: Mean = 0.8582563387651956, Std = 0.022491676348018876


	Training model: Ada


		Accuracy: Mean = 0.7381954184198223, Std = 0.022893107587604218
		F1 Score: Mean = 0.7373090209671632, Std = 0.02171327794924833
		Precision: Mean = 0.7445803721463375, Std = 0.01866350077119473
		Recall: Mean = 0.7381954184198223, Std = 0.022893107587604218
		ROC-AUC: Mean = 0.8755509527251031, Std = 0.020844944568577474


	Training model: NN


		Accuracy: Mean = 0.8311360448807854, Std = 0.03330199292909915
		F1 Score: Mean = 0.8302358477851806, Std = 0.03133485516278241
		Precision: Mean = 0.8370578067530644, Std = 0.030336178950621507
		Recall: Mean = 0.8311360448807854, Std = 0.03330199292909915
		ROC-AUC: Mean = 0.9570115495063855, Std = 0.010628757255354728


	Training model: XGBoost


		Accuracy: Mean = 0.8657550257129498, Std = 0.022485102336360703
		F1 Score: Mean = 0.8658940381574795, Std = 0.02176016156585952
		Precision: Mean = 0.8686092662098248, Std = 0.02209472315654317
		Recall: Mean = 0.8657550257129498, Std = 0.022485102336360703
		ROC-AUC: Mean = 0.966206921569554, Std = 0.008106296847018747


	Training model: LightGBM


		Accuracy: Mean = 0.8636512388966807, Std = 0.023116565939993703
		F1 Score: Mean = 0.8644460082920045, Std = 0.021805468735739013
		Precision: Mean = 0.8686343503860764, Std = 0.021138164358992573
		Recall: Mean = 0.8636512388966807, Std = 0.023116565939993703
		ROC-AUC: Mean = 0.9702605212607484, Std = 0.003644047521717108


	Training model: SVM


		Accuracy: Mean = 0.831229546517064, Std = 0.04865236374030303
		F1 Score: Mean = 0.8315224938561684, Std = 0.04880787741904113
		Precision: Mean = 0.8347335076371355, Std = 0.04858204611971036
		Recall: Mean = 0.831229546517064, Std = 0.04865236374030303
		ROC-AUC: Mean = 0.9425057730192986, Std = 0.01904031676471531


	Training model: HistGB


		Accuracy: Mean = 0.8635343618513325, Std = 0.03355145677037068
		F1 Score: Mean = 0.8644255658149176, Std = 0.03203170035844946
		Precision: Mean = 0.868011838763886, Std = 0.030486858813943415
		Recall: Mean = 0.8635343618513325, Std = 0.03355145677037068
		ROC-AUC: Mean = 0.9703779496160617, Std = 0.007608821327499804


Processing dataset: Audio Ablated 1
	Training model: KNN


		Accuracy: Mean = 0.7814633006077607, Std = 0.03542916298243898
		F1 Score: Mean = 0.7652382459883168, Std = 0.03654936756934635
		Precision: Mean = 0.7685419784870866, Std = 0.03652102835206825
		Recall: Mean = 0.7814633006077607, Std = 0.03542916298243898
		ROC-AUC: Mean = 0.9386854190231722, Std = 0.009608947217732823


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: DT


		Accuracy: Mean = 0.7208274894810659, Std = 0.03550089797766432
		F1 Score: Mean = 0.7228704339045311, Std = 0.04013325068665368
		Precision: Mean = 0.7319518599815914, Std = 0.04665930979045353
		Recall: Mean = 0.7208274894810659, Std = 0.03550089797766432
		ROC-AUC: Mean = 0.8607846296608873, Std = 0.026250849261362764


	Training model: RF


		Accuracy: Mean = 0.8094904160822814, Std = 0.03733638997265139
		F1 Score: Mean = 0.7992874548503651, Std = 0.03734066394169523
		Precision: Mean = 0.8026538124788785, Std = 0.04653436365074187
		Recall: Mean = 0.8094904160822814, Std = 0.03733638997265139
		ROC-AUC: Mean = 0.9547134291623026, Std = 0.0139850063382988


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.7705236091631603, Std = 0.03259110489186567
		F1 Score: Mean = 0.7728063800436397, Std = 0.039633292804632264
		Precision: Mean = 0.7888618771747651, Std = 0.05265473108785812
		Recall: Mean = 0.7705236091631603, Std = 0.03259110489186567
		ROC-AUC: Mean = 0.941941672850648, Std = 0.014931238485908804


	Training model: RUSBoost


		Accuracy: Mean = 0.6884057971014493, Std = 0.04274256029369755
		F1 Score: Mean = 0.698214042029804, Std = 0.053816034266204404
		Precision: Mean = 0.7187487870248184, Std = 0.06817851569963519
		Recall: Mean = 0.6884057971014493, Std = 0.04274256029369755
		ROC-AUC: Mean = 0.8933068349975081, Std = 0.02647794784833094


	Training model: Ada


		Accuracy: Mean = 0.7208508648901356, Std = 0.06684327279495518
		F1 Score: Mean = 0.7088863751229352, Std = 0.0734138718666303
		Precision: Mean = 0.7164260045004208, Std = 0.06750159217965945
		Recall: Mean = 0.7208508648901356, Std = 0.06684327279495518
		ROC-AUC: Mean = 0.9017702871942248, Std = 0.01841213627168932


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: NN


		Accuracy: Mean = 0.8744974287050022, Std = 0.04878657877383371
		F1 Score: Mean = 0.8726794012382705, Std = 0.047829759250681794
		Precision: Mean = 0.8776473991377876, Std = 0.05075354591123404
		Recall: Mean = 0.8744974287050022, Std = 0.04878657877383371
		ROC-AUC: Mean = 0.9719407311160568, Std = 0.007976875677825592


	Training model: XGBoost


		Accuracy: Mean = 0.8355539971949509, Std = 0.038702717810223955
		F1 Score: Mean = 0.8313497607398489, Std = 0.03638549154270675
		Precision: Mean = 0.8354337764222766, Std = 0.037789818597521715
		Recall: Mean = 0.8355539971949509, Std = 0.038702717810223955
		ROC-AUC: Mean = 0.9640457266494561, Std = 0.01436291923377081


	Training model: LightGBM


		Accuracy: Mean = 0.8333800841514727, Std = 0.03374461941327687
		F1 Score: Mean = 0.8255681147887687, Std = 0.031889795501629147
		Precision: Mean = 0.8381026196073169, Std = 0.03258475279302458
		Recall: Mean = 0.8333800841514727, Std = 0.03374461941327687
		ROC-AUC: Mean = 0.9598992131412347, Std = 0.012258619374056227


	Training model: SVM


		Accuracy: Mean = 0.8030388031790556, Std = 0.03169462397643682
		F1 Score: Mean = 0.7958057543138275, Std = 0.03400422725886099
		Precision: Mean = 0.8016315954402884, Std = 0.03454664224585548
		Recall: Mean = 0.8030388031790556, Std = 0.03169462397643682
		ROC-AUC: Mean = 0.942367420425497, Std = 0.01499257958377147


	Training model: HistGB


		Accuracy: Mean = 0.8420757363253857, Std = 0.03345472670594328
		F1 Score: Mean = 0.8326945505052553, Std = 0.030519909870100632
		Precision: Mean = 0.8326716494869892, Std = 0.03408887887926169
		Recall: Mean = 0.8420757363253857, Std = 0.03345472670594328
		ROC-AUC: Mean = 0.9636598126851691, Std = 0.012910291526973178


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Processing dataset: Vision Ablated 1
	Training model: KNN


		Accuracy: Mean = 0.8225338943431509, Std = 0.019899682205497468
		F1 Score: Mean = 0.8219859567672729, Std = 0.018987215404147162
		Precision: Mean = 0.8260440404426947, Std = 0.017122668952127225
		Recall: Mean = 0.8225338943431509, Std = 0.019899682205497468
		ROC-AUC: Mean = 0.9388570028583368, Std = 0.004595587148791606


	Training model: DT


		Accuracy: Mean = 0.7726741467975689, Std = 0.04416059613718496
		F1 Score: Mean = 0.7747839526897232, Std = 0.04357261515435724
		Precision: Mean = 0.7811738877536362, Std = 0.042711002339902125
		Recall: Mean = 0.7726741467975689, Std = 0.04416059613718496
		ROC-AUC: Mean = 0.8830914807227277, Std = 0.02443544864122582


	Training model: RF


		Accuracy: Mean = 0.8310659186535764, Std = 0.030102372235356924
		F1 Score: Mean = 0.8316828575682607, Std = 0.02810890371907031
		Precision: Mean = 0.8369191226612566, Std = 0.02611599672385443
		Recall: Mean = 0.8310659186535764, Std = 0.030102372235356924
		ROC-AUC: Mean = 0.9529437763482431, Std = 0.0040782370708144266


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.8094202898550724, Std = 0.02689976425541668
		F1 Score: Mean = 0.8101113784974803, Std = 0.02621162659390847
		Precision: Mean = 0.8152166044128493, Std = 0.028318363331439993
		Recall: Mean = 0.8094202898550724, Std = 0.02689976425541668
		ROC-AUC: Mean = 0.9435457138672391, Std = 0.00298525323857649


	Training model: RUSBoost


		Accuracy: Mean = 0.7098877980364657, Std = 0.047986327372545066
		F1 Score: Mean = 0.7016438358983117, Std = 0.04360157103673465
		Precision: Mean = 0.7163160509022317, Std = 0.049160527094895275
		Recall: Mean = 0.7098877980364657, Std = 0.047986327372545066
		ROC-AUC: Mean = 0.8653539631788585, Std = 0.018568803219984036


	Training model: Ada


		Accuracy: Mean = 0.6753623188405797, Std = 0.06498214890617447
		F1 Score: Mean = 0.6797818892558553, Std = 0.058616515952216716
		Precision: Mean = 0.6893451992412254, Std = 0.051058458834141805
		Recall: Mean = 0.6753623188405797, Std = 0.06498214890617447
		ROC-AUC: Mean = 0.8575215097488424, Std = 0.022983856676873828


	Training model: NN


		Accuracy: Mean = 0.7920991117344552, Std = 0.030124944508694677
		F1 Score: Mean = 0.7880345211096007, Std = 0.031233458467757477
		Precision: Mean = 0.7924231354735822, Std = 0.029932537613592616
		Recall: Mean = 0.7920991117344552, Std = 0.030124944508694677
		ROC-AUC: Mean = 0.9245053376872431, Std = 0.00954575741639197


	Training model: XGBoost


		Accuracy: Mean = 0.8073632538569424, Std = 0.010527345865242762
		F1 Score: Mean = 0.8067785130250174, Std = 0.011709467758045914
		Precision: Mean = 0.8109159798625718, Std = 0.010502753673628617
		Recall: Mean = 0.8073632538569424, Std = 0.010527345865242762
		ROC-AUC: Mean = 0.9391279409410881, Std = 0.009149820307064075


	Training model: LightGBM


		Accuracy: Mean = 0.8116175783076205, Std = 0.028409383976931917
		F1 Score: Mean = 0.811488467684903, Std = 0.027997802394250863
		Precision: Mean = 0.8186392647173524, Std = 0.029369203154386347
		Recall: Mean = 0.8116175783076205, Std = 0.028409383976931917
		ROC-AUC: Mean = 0.9483464744978394, Std = 0.0067745896781989


	Training model: SVM


		Accuracy: Mean = 0.7620149602618046, Std = 0.027255606689832793
		F1 Score: Mean = 0.7598301970113122, Std = 0.0274037645299222
		Precision: Mean = 0.7645540177529925, Std = 0.026553921108950065
		Recall: Mean = 0.7620149602618046, Std = 0.027255606689832793
		ROC-AUC: Mean = 0.896038498623754, Std = 0.012766626807254398


	Training model: HistGB


		Accuracy: Mean = 0.8332632071061242, Std = 0.029194529037259307
		F1 Score: Mean = 0.8346595908662826, Std = 0.02789713273294007
		Precision: Mean = 0.8411507334574326, Std = 0.028025490399921273
		Recall: Mean = 0.8332632071061242, Std = 0.029194529037259307
		ROC-AUC: Mean = 0.9463020395551333, Std = 0.008451052197936295


Processing dataset: Audio Ablated 2
	Training model: KNN


		Accuracy: Mean = 0.7598410472183263, Std = 0.06816737418119109
		F1 Score: Mean = 0.7486828715218272, Std = 0.0704648202790693
		Precision: Mean = 0.7572101108141789, Std = 0.06801436310563218
		Recall: Mean = 0.7598410472183263, Std = 0.06816737418119109
		ROC-AUC: Mean = 0.9211456107958224, Std = 0.021372067281955554


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: DT


		Accuracy: Mean = 0.6860682561944834, Std = 0.06710757923833022
		F1 Score: Mean = 0.6878483498546121, Std = 0.06644033527617887
		Precision: Mean = 0.700362612112724, Std = 0.06515679209851885
		Recall: Mean = 0.6860682561944834, Std = 0.06710757923833022
		ROC-AUC: Mean = 0.8285438016583363, Std = 0.03711106759833574


	Training model: RF


		Accuracy: Mean = 0.7964936886395512, Std = 0.048706040286188856
		F1 Score: Mean = 0.7903465329930274, Std = 0.047917765399231746
		Precision: Mean = 0.804329146878622, Std = 0.04943709244409699
		Recall: Mean = 0.7964936886395512, Std = 0.048706040286188856
		ROC-AUC: Mean = 0.9422012162647521, Std = 0.014925860264094658


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.7424497428705001, Std = 0.036789700780954566
		F1 Score: Mean = 0.7492403689463029, Std = 0.040362971195424716
		Precision: Mean = 0.7660767767645812, Std = 0.04289283474661299
		Recall: Mean = 0.7424497428705001, Std = 0.036789700780954566
		ROC-AUC: Mean = 0.929930677962646, Std = 0.016411870672686096


	Training model: RUSBoost


		Accuracy: Mean = 0.6732117812061712, Std = 0.03582345311803077
		F1 Score: Mean = 0.686150071260081, Std = 0.04688088751371725
		Precision: Mean = 0.708767607061938, Std = 0.062080604133016824
		Recall: Mean = 0.6732117812061712, Std = 0.03582345311803077
		ROC-AUC: Mean = 0.88017362570169, Std = 0.020969635052456425


	Training model: Ada


		Accuracy: Mean = 0.6862786348761104, Std = 0.08675900141402472
		F1 Score: Mean = 0.6770489222271018, Std = 0.09189499259479417
		Precision: Mean = 0.6809764322320142, Std = 0.09158098513274204
		Recall: Mean = 0.6862786348761104, Std = 0.08675900141402472
		ROC-AUC: Mean = 0.8853033292269421, Std = 0.0223681481478653


	Training model: NN


		Accuracy: Mean = 0.8030154277699861, Std = 0.031732718334844395
		F1 Score: Mean = 0.7970695999888594, Std = 0.03575106867868484
		Precision: Mean = 0.8082888796842678, Std = 0.029012440125267425
		Recall: Mean = 0.8030154277699861, Std = 0.031732718334844395
		ROC-AUC: Mean = 0.952488630980459, Std = 0.014826345340369598


	Training model: XGBoost


		Accuracy: Mean = 0.7878447872837774, Std = 0.03692378018425561
		F1 Score: Mean = 0.785626645567128, Std = 0.03657589180828259
		Precision: Mean = 0.7897541495829147, Std = 0.03713712009888141
		Recall: Mean = 0.7878447872837774, Std = 0.03692378018425561
		ROC-AUC: Mean = 0.938126691175807, Std = 0.01755950612201127


	Training model: LightGBM


		Accuracy: Mean = 0.8095605423094904, Std = 0.04237940390461628
		F1 Score: Mean = 0.8039233629393848, Std = 0.04073166343568151
		Precision: Mean = 0.8130715531254626, Std = 0.04222946636643777
		Recall: Mean = 0.8095605423094904, Std = 0.04237940390461628
		ROC-AUC: Mean = 0.9433901852367426, Std = 0.017472877674367074


	Training model: SVM


		Accuracy: Mean = 0.7510986442262739, Std = 0.039282275834153106
		F1 Score: Mean = 0.7489953360071894, Std = 0.04207408821444491
		Precision: Mean = 0.75591936883992, Std = 0.04502615611170873
		Recall: Mean = 0.7510986442262739, Std = 0.039282275834153106
		ROC-AUC: Mean = 0.926360852781088, Std = 0.01969223971590549


	Training model: HistGB


		Accuracy: Mean = 0.8008181393174381, Std = 0.05773235798263222
		F1 Score: Mean = 0.7961918144226414, Std = 0.0577442715039489
		Precision: Mean = 0.8037010657359982, Std = 0.05965773529151661
		Recall: Mean = 0.8008181393174381, Std = 0.05773235798263222
		ROC-AUC: Mean = 0.9445992348364547, Std = 0.022309883615974756


Processing dataset: Audio Ablated 3
	Training model: KNN


		Accuracy: Mean = 0.7814633006077607, Std = 0.03542916298243898
		F1 Score: Mean = 0.7652382459883168, Std = 0.03654936756934635
		Precision: Mean = 0.7685419784870866, Std = 0.03652102835206825
		Recall: Mean = 0.7814633006077607, Std = 0.03542916298243898
		ROC-AUC: Mean = 0.9386854190231722, Std = 0.009608947217732823


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: DT


		Accuracy: Mean = 0.7121552127162225, Std = 0.040077028073924534
		F1 Score: Mean = 0.7169587909890274, Std = 0.04310618016332177
		Precision: Mean = 0.7302821603015482, Std = 0.04827242819994835
		Recall: Mean = 0.7121552127162225, Std = 0.040077028073924534
		ROC-AUC: Mean = 0.8605141853852303, Std = 0.029275788006380574


	Training model: RF


		Accuracy: Mean = 0.7986909770920991, Std = 0.041378871566051786
		F1 Score: Mean = 0.7863607967264634, Std = 0.039753601566127755
		Precision: Mean = 0.7795771545981172, Std = 0.04067550341494826
		Recall: Mean = 0.7986909770920991, Std = 0.041378871566051786
		ROC-AUC: Mean = 0.9545426058888825, Std = 0.012841790826075873


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: BalancedRF


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:546: FutureWarning: The default of `sampling_strategy` will change from `'auto'` to `'all'` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `'all'` to silence this warning and adopt the future behaviour.
  warn(
/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/imblearn/ensemble/_forest.py:558: FutureWarning: The default of `replacement` will change from `False` to `True` in version 0.13. This change will follow the implementation proposed in the original paper. Set to `True` to silence this warning and adopt the future behaviour.
  warn(


		Accuracy: Mean = 0.7770219728845253, Std = 0.033356340172249005
		F1 Score: Mean = 0.7800037110256273, Std = 0.03825552478409238
		Precision: Mean = 0.7924552576238093, Std = 0.04439263838782084
		Recall: Mean = 0.7770219728845253, Std = 0.033356340172249005
		ROC-AUC: Mean = 0.9436090641382322, Std = 0.015767153053108236


	Training model: RUSBoost


		Accuracy: Mean = 0.6949509116409537, Std = 0.04072805328332794
		F1 Score: Mean = 0.702857963422669, Std = 0.05009468539910493
		Precision: Mean = 0.7251076859381644, Std = 0.06563025763626214
		Recall: Mean = 0.6949509116409537, Std = 0.04072805328332794
		ROC-AUC: Mean = 0.8827853627054987, Std = 0.025988108616433366


	Training model: Ada


		Accuracy: Mean = 0.7208508648901356, Std = 0.06684327279495518
		F1 Score: Mean = 0.7088863751229352, Std = 0.0734138718666303
		Precision: Mean = 0.7164260045004208, Std = 0.06750159217965945
		Recall: Mean = 0.7208508648901356, Std = 0.06684327279495518
		ROC-AUC: Mean = 0.9017702871942248, Std = 0.01841213627168932


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


	Training model: NN


		Accuracy: Mean = 0.8722533894343151, Std = 0.03598803462505809
		F1 Score: Mean = 0.8697465284102799, Std = 0.03595215066539302
		Precision: Mean = 0.8781860908022312, Std = 0.03272921939457558
		Recall: Mean = 0.8722533894343151, Std = 0.03598803462505809
		ROC-AUC: Mean = 0.9742765238404203, Std = 0.013689349614897841


	Training model: XGBoost


		Accuracy: Mean = 0.8355539971949509, Std = 0.038702717810223955
		F1 Score: Mean = 0.8313497607398489, Std = 0.03638549154270675
		Precision: Mean = 0.8354337764222766, Std = 0.037789818597521715
		Recall: Mean = 0.8355539971949509, Std = 0.038702717810223955
		ROC-AUC: Mean = 0.9640457266494561, Std = 0.01436291923377081


	Training model: LightGBM


		Accuracy: Mean = 0.8333800841514727, Std = 0.03374461941327687
		F1 Score: Mean = 0.8255681147887687, Std = 0.031889795501629147
		Precision: Mean = 0.8381026196073169, Std = 0.03258475279302458
		Recall: Mean = 0.8333800841514727, Std = 0.03374461941327687
		ROC-AUC: Mean = 0.9598992131412347, Std = 0.012258619374056227


	Training model: SVM


		Accuracy: Mean = 0.7988312295465171, Std = 0.04076160642853108
		F1 Score: Mean = 0.7895226430264273, Std = 0.03707540716888585
		Precision: Mean = 0.7902059274666632, Std = 0.02920807962534188
		Recall: Mean = 0.7988312295465171, Std = 0.04076160642853108
		ROC-AUC: Mean = 0.9456246242217985, Std = 0.012626844578587194


	Training model: HistGB


		Accuracy: Mean = 0.8420757363253857, Std = 0.03345472670594328
		F1 Score: Mean = 0.8326945505052553, Std = 0.030519909870100632
		Precision: Mean = 0.8326716494869892, Std = 0.03408887887926169
		Recall: Mean = 0.8420757363253857, Std = 0.03345472670594328
		ROC-AUC: Mean = 0.9636598126851691, Std = 0.012910291526973178


/home/chenlequn/anaconda3/envs/torch/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


In [14]:
%store datasets
%store le
%store Y_encoded

Stored 'datasets' (dict)
Stored 'le' (LabelEncoder)
Stored 'Y_encoded' (ndarray)
